### PR TITLE
Assigning literal values to NUM_DIGITAL_PINS and NUM_ANALOG_INPUTS

### PR DESCRIPTION
**Pull request by lacklustrlabs**
It turned out that I needed the fix of #140 for a pet project of mine. 
So here are the changes I made.

I've compiled every board with this .ino:
```
#if NUM_DIGITAL_PINS > 4
#else
#error "NUM_DIGITAL_PINS not fixed"
#endif

#if NUM_ANALOG_INPUTS > 1
#else
#error "NUM_ANALOG_INPUTS not fixed"
#endif

void setup() {
}

void loop() {
} 
```

### CODE REVIEW

**Comment by fpistm** [comment]
Thanks @lacklustrlabs 
First check seems ok. I need to review it more deeper😉
It is the first part of the #140 which require also Ax pins definition review

**Comment by lacklustrlabs** [comment]
imho it would be easiest to add the first PIN_AX variant:
```
#if NUM_ANALOG_INPUTS > 0
#define PIN_A0 A0
#endif
#if NUM_ANALOG_INPUTS > 1
#define PIN_A1 A1
#endif
...
#if NUM_ANALOG_INPUTS > 18
#define PIN_A18 A18
#endif
```
I can do that, but the question is where do you want it?

**Comment by fpistm** [comment]
Thanks @lacklustrlabs 
I have to check that before doing any change for Ax definition. I would like check how other variant do it.
Anyway, I could merge your PR before doing that.
I have only to review code and made some tests. (firmata, ....)

File: cores/arduino/pins_arduino_var.h [hunk 1]
```diff
@@ -25,14 +25,16 @@
 // Include board variant
 #include "variant.h"
 
+// Avoid pins number misalignment
+_Static_assert(NUM_DIGITAL_PINS==PEND, "NUM_DIGITAL_PINS and PEND differ!");
+_Static_assert(NUM_ANALOG_INPUTS==(AEND-A0), "NUM_DIGITAL_PINS and (AEND-A0) differ!");
```
Comment (fpistm) [inline cores/arduino/pins_arduino_var.h]: Sorry made a wrong copy/paste
"~NUM_DIGITAL_PINS~NUM_ANALOG_INPUTS and (AEND-A0) differ!"

Comment (lacklustrlabs) [reply cores/arduino/pins_arduino_var.h]: No problem :)

**Pull request merged**